### PR TITLE
shows confirm dailog before deleting bookmark 

### PR DESCRIPTION
**Pull request by 4shutosh**
For  #1798 for bookmarks
* Shows a dailog for confirmation before deleting bookmarks.

**Screenshots** 
Webp net-resizeimage (6)

### CODE REVIEW

**Comment by 4shutosh** [comment]
Implemented Confirmation Dialog for delelting bookmarks
Please review!

**Comment by kelson42** [comment]
Both bookmarks and hopistory dialogs should be implemented.

**Comment by srv-twry** [comment]
@4shutosh Hi,

Actually I had expressed my willingness to work on this issue in the description itself which you probably missed. I had already started working on this.

I'll let this one go since it's a small feature but request you to kindly pay attention to detail in order to avoid conflicts like this in the future.

**Comment by 4shutosh** [comment]
> @4shutosh Hi,
> 
> Actually I had expressed my willingness to work on this issue in the description itself which you probably missed. I had already started working on this.
> 
> I'll let this one go since it's a small feature but request you to kindly pay attention to detail in order to avoid conflicts like this in the future.
Sorry for inconvenience @srv-twry this won't happen again.
If you want I'll close this PR.
I kindly request you to give suggestions on the code for the same and help to fix the bug.

**Comment by 4shutosh** [comment]
> Both bookmarks and history dialogs should be implemented.

Done.
Please Review!

File: core/src/main/java/org/kiwix/kiwixmobile/core/bookmark/BookmarksActivity.java [hunk 1]
```diff
@@ -195,11 +196,21 @@ public boolean onOptionsItemSelected(MenuItem item) {
       onBackPressed();
       return true;
     } else if (itemId == R.id.menu_bookmarks_clear) {
-      presenter.deleteBookmarks(new ArrayList<>(allBookmarks));
-      allBookmarks.clear();
-      bookmarksList.clear();
-      bookmarksAdapter.notifyDataSetChanged();
-      Toast.makeText(this, R.string.all_bookmarks_cleared_toast, Toast.LENGTH_SHORT).show();
+      new AlertDialog.Builder(this)
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/bookmark/BookmarksActivity.java]: All dialog usage should be through the DialogShower class

File: core/src/main/java/org/kiwix/kiwixmobile/core/bookmark/BookmarksActivity.java [hunk 2]
```diff
@@ -195,11 +196,21 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+              .setTitle(getResources().getString(R.string.clear_all_bookmark_dialog_title))
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/bookmark/BookmarksActivity.java]: Activities have a method `getString` so you don't need to use `getResources` to get a string but also there is a version of `setTitle` that takes an int

File: core/src/main/java/org/kiwix/kiwixmobile/core/bookmark/BookmarksActivity.java [hunk 3]
```diff
@@ -195,11 +196,21 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+              .setMessage(getResources().getString(R.string.clear_bookmarks_dailog))
+              .setPositiveButton(android.R.string.yes, (dialog, which) -> {
+                presenter.deleteBookmarks(new ArrayList<>(allBookmarks));
+                allBookmarks.clear();
+                bookmarksList.clear();
+                bookmarksAdapter.notifyDataSetChanged();
+                Toast.makeText(this, R.string.all_bookmarks_cleared_toast, Toast.LENGTH_SHORT).show();
+              })
+              .setNegativeButton(android.R.string.no, (dialog, which) -> {
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/bookmark/BookmarksActivity.java]: Setting a null clickListener is a valid option for this parameter

**Comment by macgills** [comment]
Closing as per my comment on the ticket

**Pull request closed without merging**